src/include/homestore/btree/detail/btree_mutate_impl.ipp

```diff
@@ -236,9 +236,9 @@ btree_status_t Btree< K, V >::check_split_root(ReqT& req) {
 
     ret = split_node(root, child_node, root->total_entries(), &split_key, req.m_op_context);
     if (ret != btree_status_t::success) {
+        on_root_changed(child_node, root, req.m_op_context); // Revert it back
```

Comment:
please explain why do we need to have child_node for revert the root id in superblk?

Reply:
The child_node here is the original root node - and we're reverting what we're doing at L230.

Reply:
ok . the meta buffer is link to root(or new root) why do we need to pass the new argument?
It is like meta->root (whichever it is)
Then in merge and split with create the rest of DAG

Reply:
please elaborate for me the necessity of this change with a simple counterexample